assets/src/scripts/components/hydrograph/date-controls.spec.js

```diff
@@ -0,0 +1,246 @@
+import { select } from 'd3-selection';
+import {Actions, configureStore} from '../../store';
+import {attachToNode} from './index';
```

Comment:
You should test the exported function from the date-controls module (drawDateRangeControls) rather than calling attachToNode. You should be able to modify the tests below and potentially add additional tests because may only need to mock the part of the state that you need need to test the rendering code with.